src/catkin_pkg/packages.py

```diff
@@ -129,25 +129,35 @@ def find_packages_allowing_duplicates(basepath, exclude_paths=None, exclude_subs
     if not data:
         return {}
 
-    if len(data) < 100:
+    parallel = len(data) > 100
+    if parallel:
+        try:
+            parser = _PackageParser(warnings is not None)
```

Comment:
This line shouldn't raise and therefore shouldn't be in the try/except block. It should be moved in to `parallel` block below.